proxy.go

```diff
@@ -1,96 +1,111 @@
 package solidproxy
 
 import (
-	"crypto/tls"
+	"io/ioutil"
 	"net/http"
 	"net/url"
-	"sync"
 	"time"
-
-	"github.com/elazarl/goproxy"
-)
-
-var (
-	proxy    = goproxy.NewProxyHttpServer()
-	cookies  = map[string]map[string][]*http.Cookie{}
-	cookiesL = new(sync.RWMutex)
 )
 
-func InitProxy(conf *ServerConfig) {
-	proxy.OnResponse().DoFunc(func(r *http.Response, ctx *goproxy.ProxyCtx) *http.Response {
-		// CORS
-		r.Header.Set("Access-Control-Allow-Credentials", "true")
-		r.Header.Set("Access-Control-Expose-Headers", "User, Triples, Location, Link, Vary, Last-Modified, Content-Length")
-		r.Header.Set("Access-Control-Max-Age", "60")
+func ProxyHandler(w http.ResponseWriter, req *http.Request) {
+	Logger.Println("New request from:", req.RemoteAddr, "for URI:", req.URL.String())
 
-		// Retry with server credentials if authentication is required
-		if r.StatusCode == 401 {
-			// for debugging
-			defer TimeTrack(time.Now(), "Fetching")
-			var resp *http.Response
-			var client *http.Client
-			var solutionMsg string
-			req, err := http.NewRequest("GET", ctx.Req.URL.String(), ctx.Req.Body)
-			req.Header.Add("On-Behalf-Of", conf.User)
-			// Retry the request
-			if len(cookies[conf.User]) > 0 { // Use existing cookie
-				req.AddCookie(cookies[conf.User][ctx.Req.Host][0])
-				// Create the client
-				client = &http.Client{
-					Transport: &http.Transport{
-						TLSClientConfig: &tls.Config{
-							InsecureSkipVerify: conf.InsecureSkipVerify,
-						},
-					},
-				}
-				solutionMsg = "Retrying with cookies"
-			} else { // Using WebIDTLS client
-				client = webidTlsClient
-				solutionMsg = "Retrying with WebID-TLS"
-			}
-			resp, err = client.Do(req)
-			if err != nil {
-				Logger.Fatal("GET:", err)
-			}
-			// Store cookies per user and request host
-			if len(resp.Cookies()) > 0 {
-				cookiesL.Lock()
-				// Should store cookies based on domain value AND path from cookie
-				cookies[conf.User] = map[string][]*http.Cookie{}
-				cookies[conf.User][ctx.Req.Host] = resp.Cookies()
-				Logger.Printf("Cookies: %+v\n", cookies)
-				cookiesL.Unlock()
-			}
-			Logger.Println("Resource "+ctx.Req.URL.String(),
-				"requires authentication (HTTP 401).", solutionMsg,
-				"resulted in HTTP", resp.StatusCode)
-			return resp
-		}
-		Logger.Println("Received data with status HTTP", r.StatusCode)
-		return r
-	})
-}
+	user := req.Header.Get("User")
+	// override if we have specified a user in config
+	if len(userWebID) > 0 {
+		user = userWebID
+	}
 
-func ProxyHandler(w http.ResponseWriter, req *http.Request) {
-	toProxy := req.FormValue("uri")
-	if len(toProxy) == 0 {
+	uri := req.FormValue("uri")
+	if len(uri) == 0 {
+		msg := "No URI was provided to the proxy!"
+		Logger.Println(msg, req.URL.String())
 		w.WriteHeader(500)
```

Comment:
Yeah I think this should be a `400`. I don't think a server should really ever explicitly return a `500`.